game/hud/src/components/HUD/index.tsx

```diff
@@ -94,41 +102,47 @@ class HUD extends React.Component<HUDProps, HUDState> {
 
     console.log(e.nativeEvent.deltaY);
 
+    const pos: Position = this.props.layout.widgets[name];
+
     if (e.nativeEvent.deltaY < 0) {
       this.props.dispatch(savePosition(name, {
-        x: this.props.layout.widgets[name].x,
-        y: this.props.layout.widgets[name].y,
-        scale: this.props.layout.widgets[name].scale - factor
+        x: pos.x,
+        y: pos.y,
+        width: pos.width,
+        height: pos.height,
+        scale: pos.scale - factor
       }));
     } else {
       this.props.dispatch(savePosition(name, {
-        x: this.props.layout.widgets[name].x,
-        y: this.props.layout.widgets[name].y,
-        scale: this.props.layout.widgets[name].scale + factor
+        x: pos.x,
+        y: pos.y,
+        width: pos.width,
+        height: pos.height,
+        scale: pos.scale + factor
       }));
     }
   }
 
   draggableWidget = (name: string, widgets: any, Widget: any, containerClass: string, props?: any) => {
-    const w = widgets[name];
+    const pos: Position = widgets[name];
```

Comment:
validate objects and throw an appropriate exception, doing so as a general rule will help us diagnose issues especially when reported by users/hard to repro errors